pkg/backup/scheduler.go

```diff
@@ -0,0 +1,112 @@
+package backup
+
+import (
+	"context"
+	"fmt"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	logger "github.com/rs/zerolog/log"
+)
+
+var log = logger.With().Str("component", "backup").Logger()
+
+// Scheduler executes backups at a regular interval.
+type Scheduler struct {
+	Frequency      int // in minutes
+	NotificationCh chan bool
```

Comment:
The `NotificationCh` is used to notify them that the backup process is finished for the client. This is used only in the test right now but can be leveraged by other occasions in the future.